README.md

```diff
@@ -29,7 +29,15 @@ include ::profile_update_os
 
 ## Usage
 
-The goal is that no paramters are required to be set. The default paramters should work for most NCSA deployments out of the box.
+### kpatch
+
+Currently only supported on Redhat hosts (centos/rocky do not appear to publish kpatch-patch packages)
+
+kpatch allows patching for kernel vulnerabilities without rebooting (host stays on current kernel version but gets a package which makes the old kernel safe).
+
+Set `profile_update_os::kpatch::enabled: true` to have host install kpatch package and start the kpatch service. This will not mitigate anything, but will make the host ready to install a kpatch-patch if needed
+
+Set `profile_update_os::kpatch::install_kpatch_patches: true` to have the host install the kpatch-patch for the current kernel (kpatch installs an "empty" kpatch-patch if there are no existing kpatch-patches available)
```

Comment:
So enabling `profile_update_os::kpatch::install_kpatch_patches: true` would install the relevant patch on the next Puppet run after the kpatch-patch for the current kernel is made available in the Yum repos. Is that correct?

That probably makes sense, but perhaps there could/should be some sort of option to schedule that during a predefined time of day/week, similar to how kernel updates in this module work.

Reply:
Correct, once `profile_update_os::kpatch::install_kpatch_patches: true` is set it will start updating the kpatch-patch to the most recent version it can find.

I'm wondering what the ability to schedule the kpatch-patch would buy us. For the kernel patching it makes sense because that will require a reboot so it's nice to schedule those, but the kpatch-patches don't need a reboot so wondering why we would want to schedule them.

Reply:
While I generally agree, my concern is that using kpatch is new to us and we don't have much experience with it production.

Until we have used it for a while and are sure there are no issues, I'd feel more comfortable if it only updated during work days rather than after hours.

Reply:
Ah I see. The defaults have `profile_update_os::kpatch::install_kpatch_patches: false`. The way I envision us using this is that we leave that at false until IRST gives us a heads up that there is something we need to patch ASAP. Once we verify there is a kpatch-patch that covers the issue we intentionally flip it to `true`. Then once systems are on a kernel that have the real patch we flip it back to `false`.

If that work-flow makes sense I can add those details to the README

On systems that use repo snapshots the rollout would be controlled in a precise way. I suppose systems that don't use repo snapshots could have a window where we are applying kpatch-patch_v1 for CVE_1, and then while waiting to get everything on the new fixed kernel kpatch-patch_v2 for CVE_2 comes out.

To alleviate that concern, maybe instead of adding a scheduling option we add an option to either install the latest version, or specifically install a targeted release (look like version # is either 0 or 1, they only bump up the release # when new CVEs need patching).